lib/CL/devices/common.h

```diff
@@ -26,7 +26,7 @@
 #define POCL_COMMON_H
 
 #include "pocl_cl.h"
-#include "dev_image.h"
+#include "_kernel_c.h"
```

Comment:
_kernel_c.h should be included only from the kernel builtin function definitions (lib/kernel). It doesn't seem to build with gcc at the moment due to #if __has_attribute(__always_inline__) which is recognized by Clang but not gcc 4.8. Can you revert this part in such a way that the dev_image.h has only the image struct which _kernel_c.h includes?